channels/rdpdr/client/rdpdr_main.c

```diff
@@ -468,17 +468,21 @@ static UINT handle_hotplug(rdpdrPlugin* rdpdr)
 			if (!drive)
 			{
 				WLog_ERR(TAG, "calloc failed!");
-				return CHANNEL_RC_NO_MEMORY;
+				error = CHANNEL_RC_NO_MEMORY;
+				goto cleanup;
 			}
 
 			drive->Type = RDPDR_DTYP_FILESYSTEM;
 
-			drive->Path = _strdup(dev_array[i].path);
+			drive->Path = dev_array[i].path;
+			dev_array[i].path = NULL;
+
 			if (!drive->Path)
 			{
 				WLog_ERR(TAG, "_strdup failed!");
```

Comment:
@MartinHaimberger No, the ```_strdup``` just duplicated a string cleaned up later on. Setting the source to NULL prevents that and so no duplication is necessary.